### ISSUE TITLE
New release of espeak

### ISSUE THREAD

**Issue by feerrenrut** [espeak-ng/espeak-ng#857]
As per: #792

Over at nvaccess/nvda we have started our 2021.1 release process by [updating the version of espeak-ng](https://github.com/nvaccess/nvda/pull/11928) included. We selected the latest espeak-ng commit (at the time) and would love it if an espeak-ng release process could be started that we could use for the final release.

We have updated to espeak-ng commit 82d5b7b04488412845101851f36da6953cac4378F
The NVDA 2021.1 release will be made in the first quarter of 2021.

**Comment by jaacoppi** [comment]
@rhdunn, @valdisvi, I think we  should  make a 1.51 release. It's been more than a year and >200 commits.

**Comment by rhdunn** [comment]
Release checklist:
- [ ] #824 -- This should be fixed or reverted given that it affects NVDA.
- [ ] check/ensure all languages install correctly from the Windows installer (i.e. they have entries in the WiX file with GUIDs that are unique to each file/component).

**Comment by feerrenrut** [comment]
@rhdunn feel free to tag me, or create an issue on https://github.com/nvaccess/nvda if you'd like us to update espeak to test further. Is there a branch we should track for the release?

**Comment by jaacoppi** [comment]
Are we ready to make a new release? e6880b27d and a0b0f58cf should have fixed the windows installer.

I don't know what it takes to increase the version number. I guess the comments in 875ef11d77 should be taken into account. What else?

**Comment by valdisvi** [comment]
For PC build:
b702b039
85a9e553
For Android build:
23a3d121

**Comment by seanbudd** [comment]
Hi again,

Over at nvaccess/nvda we have now started our 2021.2 release process, and we've updated the version of espeak-ng again for testing on alpha. We would love it if we can use a new release of espeak when we finalise 2021.2. Especially if we can confirm the latest commit with #950 fixed.

**Comment by seanbudd** [comment]
We are using espeak commit `aafd2e720582d7ccdadef675ce656903bc775c1d` for our 2021.1 release.

**Comment by williamh** [comment]
Hi,

Can you please give us an update on what needs to be done before a new release can be tagged?

Thanks,

William

**Comment by jaacoppi** [comment]
The Windows installer lacks support for many newly added languages. We need 
1) a script that checks for missing languages in the Windows installer
2) add those missing languages either manually or with a script.

People with Windows development experience are encouraged to help.

**Comment by kloczek** [comment]
Installer of the binaries is not immanent part of the any project source code.
That part can be developed/finished after actual source code release.

**Comment by seanbudd** [comment]
Hi again,

Over at nvaccess/nvda we have now started our 2021.3 release process, and we've updated the version of espeak-ng to ``74068b91bcd578bd7030a7a6cde2085114b79b44`` for testing on alpha. We would love it if we can use a new release of espeak when we finalise 2021.3.

**Comment by alex19EP** [comment]
it would be nice if new version of pcaudiolib be tagged.
ref: espeak-ng/pcaudiolib#7

**Comment by ryandesign** [comment]
> it would be nice if new version of pcaudiolib be tagged.

It was. pcaudiolib 1.2 was released July 30, and then re-released today.

**Comment by valdisvi** [comment]
I plan to prepare new release of eSpeak NG at the end of 2021.

**Comment by MarcroSoft** [comment]
Hi,
I don't know if this is the right place to put this, I can't compile the master branch on windows, but I can compile 1.50.
I am having some unresolved externals like
Error	LNK1120	4 unresolved externals	libespeak-ng	D:\espeak-ng\src\windows\x64\Debug\libespeak-ng.dll	1	
Error	LNK2019	unresolved external symbol LookupSoundicon referenced in function AnnouncePunctuation	libespeak-ng	D:\espeak-ng\src\windows\readclause.obj	1	
Error	LNK2019	unresolved external symbol LoadSoundFile2 referenced in function ProcessSsmlTag	libespeak-ng	D:\espeak-ng\src\windows\ssml.obj	1	
Error	LNK2001	unresolved external symbol n_soundicon_tab	libespeak-ng	D:\espeak-ng\src\windows\synthdata.obj	1	
Error	LNK2001	unresolved external symbol n_soundicon_tab	libespeak-ng	D:\espeak-ng\src\windows\synthesize.obj	1	
Error	LNK2001	unresolved external symbol soundicon_tab	libespeak-ng	D:\espeak-ng\src\windows\synthdata.obj	1	
Error	LNK2001	unresolved external symbol soundicon_tab	libespeak-ng	D:\espeak-ng\src\windows\synthesize.obj	1	
And that is not present in 1.50.
Please let me know if you need some more information.
Thanks,
Marc

**Comment by jaacoppi** [comment]
> Hi, I don't know if this is the right place to put this, I can't compile the master branch on windows, but I can compile 1.50. I am having some unresolved externals like Error LNK1120 4 unresolved externals libespeak-ng D:\espeak-ng\src\windows\x64\Debug\libespeak-ng.dll 1 Error LNK2019 unresolved external symbol LookupSoundicon referenced in function AnnouncePunctuation libespeak-ng D:\espeak-ng\src\windows\readclause.obj 1 Error LNK2019 unresolved external symbol LoadSoundFile2 referenced in function ProcessSsmlTag libespeak-ng D:\espeak-ng\src\windows\ssml.obj 1 Error LNK2001 unresolved external symbol n_soundicon_tab libespeak-ng D:\espeak-ng\src\windows\synthdata.obj 1 Error LNK2001 unresolved external symbol n_soundicon_tab libespeak-ng D:\espeak-ng\src\windows\synthesize.obj 1 Error LNK2001 unresolved external symbol soundicon_tab libespeak-ng D:\espeak-ng\src\windows\synthdata.obj 1 Error LNK2001 unresolved external symbol soundicon_tab libespeak-ng D:\espeak-ng\src\windows\synthesize.obj 1 

Those functions are in a new file soundicon.c, headers in soundicon.h. For some reason your compiler doesn't find the header. Could you try a clean build?

The includes compile on Linux and look ok:
```
 grep soundicon.h *.c
readclause.c:#include "soundicon.h"               // for LookupSoundIcon
soundicon.c:#include "soundicon.h" 
ssml.c:#include "soundicon.h"               // for LoadSoundFile2
synthdata.c:#include "soundicon.h"               // for soundicon_tab
synthesize.c:#include "soundicon.h"               // for soundicon_tab, n_soundicon

```

**Comment by MarcroSoft** [comment]
Hi,
What do you mean with a clean build? I didn't mix the master and 1.50.
Thanks,
Marc

**Comment by PanderMusubi** [comment]
Hi all, a friendly reminder for a new release which will be welcomed by many. Thanks.

**Comment by alex19EP** [comment]
@valdisvi what about new release?

can I help with something?

**Comment by valdisvi** [comment]
There were two general issues to postpone release:
1. somebody, who is able and willing to work on Windows installer should review [issues with Windows installer](https://github.com/espeak-ng/espeak-ng/issues?q=is%3Aissue+is%3Aopen+Windows+installer) and prepare *.msi file,
2. somebody, who is able and willing should review [issue with Mandarin](https://github.com/espeak-ng/espeak-ng/issues/1044).

If for most of eSpeak NG users these issues are not important, I can prepare package for Linux based systems.

**Comment by alex19EP** [comment]
Ok I will try to look at 1

**Comment by alex19EP** [comment]
hello @valdisvi I started work on windows projects.

the question about wix installer is: do we really need sutch granularity in the choice of languages to install?

the folder with all the espeak-ng data is 13M, so I see no reason why we should not install all the data at once. especially since it will greatly simplify the installation files.

**Comment by valdisvi** [comment]
I agree, that it could be simplified.

**Comment by alex19EP** [comment]
ok I think I done. you can merge my PR create new tag and then I will build and send created MSI.

**Comment by valdisvi** [comment]
@alex19EP, I have created release: https://github.com/espeak-ng/espeak-ng/releases/tag/1.51 please use it to create MSI files (and notify me when they are prepared).
Thanks!

**Comment by alex19EP** [comment]
@valdisvi hello pleas get artifacts from: https://pkgbuild.com/~alex19ep/other/espeak-ng/msi/

**Comment by valdisvi** [comment]
Thanks @alex19EP! I added them to https://github.com/espeak-ng/espeak-ng/releases/tag/1.51

**Comment by PanderMusubi** [comment]
Please, upvote https://bugs.launchpad.net/ubuntu/+source/espeak-ng/+bug/1967857 and create an update request for Debian too.

**Comment by sthibaul** [comment]
Debian is already up to date in sid ;)
